Source/JavaScriptCore/runtime/JSStringJoiner.cpp

```diff
@@ -165,6 +165,55 @@ static inline String joinStrings(const JSStringJoiner::Entries& strings, std::sp
     return result;
 }
 
+template<typename OutputCharacterType, typename SeparatorCharacterType>
+static inline String joinStrings(JSGlobalObject* globalObject, const WriteBarrier<Unknown>* strings, unsigned size, std::span<const SeparatorCharacterType> separator, unsigned joinedLength)
+{
+    VM& vm = globalObject->vm();
+    auto scope = DECLARE_THROW_SCOPE(vm);
+
+    ASSERT(joinedLength);
+
+    std::span<OutputCharacterType> data;
+    String result = StringImpl::tryCreateUninitialized(joinedLength, data);
+    if (UNLIKELY(result.isNull())) {
+        throwOutOfMemoryError(globalObject, scope);
+        return { };
+    }
+
+    switch (separator.size()) {
+    case 0: {
```

Comment:
Nit: why a switch over an `if ... else`?

Reply:
Not a strong preference, but I was trying to mimic the structure of the more complex version of `joinStrings` above.